Utility/Utility.Public/Include/Utility/SafeQueue.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <condition_variable>
+#include <deque>
+#include <mutex>
+#include <thread>
+
+namespace Utility
+{
+template <typename T>
+/**
+ * @brief The tamplate SafeQueue class
+ */
+class SafeQueue
+{
+private:
+    std::mutex              _ScopedMutex, _UniqueMutex;
+    std::condition_variable _Block;
+
+    std::deque<T> _RawQueue;
+
+public:
+    /// Default SafeQueue constructor
+    SafeQueue() = default;
+
+    /// SafeQueue copy constructor delete
+    SafeQueue(const SafeQueue<T>& other) = delete;
+
+    /// SafeQueue destructor for clear method
+    ~SafeQueue() { clear(); }
+
+    /// Method for access to front value
+    const T& front()
+    {
+        std::scoped_lock scopedLock(_ScopedMutex);
+        return _RawQueue.front();
+    }
+
+    /// Method for access to back value
+    const T& back()
+    {
```

Comment:
In there too.

Reply:
Here I would move implementation in cpp too